src/output_updater_cuda.cu

```diff
@@ -52,6 +52,66 @@ __global__ void update_delta_z_cuda(float const *mu_a, float const *var_a,
     }
 }
 
+__global__ void update_delta_z_cuda_noise(float const *mu_a, float const *var_a,
+                                          float const *jcb, float const *obs,
+                                          int size, float *delta_mu,
+                                          float *delta_var) {
+    const float zero_pad = 0.0f;
+
+    int col = blockIdx.x * blockDim.x + threadIdx.x;
+
+    if (col % 2 == 0) {
+        // Need to pass through exponential activantion function in the update
+        // class in CUDA. To do: just use the exponential activation function
+        // from the activation class.
+        float mu_V2_bar_tilde = expf(mu_a[col + 1] + 0.5f * var_a[col + 1]);
```

Comment:
This is not idel where we need to reuse the activation function here...I am not sure why you couldn't do the same here as CPU version